cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -149,11 +158,16 @@ func (p *Planner) NodesToDelete() (empty, needDrain []*apiv1.Node) {
 		p.context.AutoscalingOptions.MaxScaleDownParallelism)
 	for _, nodeToRemove := range nodesToRemove {
 		if len(nodeToRemove.PodsToReschedule) > 0 {
-			needDrain = append(needDrain, nodeToRemove.Node)
+			if nodeToRemove.IsRisky {
```

Comment:
This check is too late: `GetNodesToRemove` above will already drop some nodes, so putting risky nodes at the end needs to happen before that.

Reply:
Thanks!